src/cleanfigure.m

```diff
@@ -342,109 +422,124 @@ function simplifyLine(meta, handle, targetResolution)
+        mask      = pixelate(xData, yData, xToPix, yToPix);
+        id_remove = find(mask==0);
+    else
+        % Get the width of a pixel
+        xPixelWidth = 1/xToPix;
+        yPixelWidth = 1/yToPix;
+        tol = min(xPixelWidth,yPixelWidth);
+
+        % Split up lines which are seperated by NaNs
+        inan   = isnan(xData) | isnan(yData);
+        df     = diff([false; ~inan; false]);
+        pstart = find(df == 1);
+        pend   = find(df == -1)-1;
+        nlines = numel(pstart);
+
+        % Count the number of data points in the previous segments
+        countData = 0;
+        for ii = 1:nlines
+            % Actual data that inherits the simplifications
+            x = xData(pstart(ii):pend(ii));
+            y = yData(pstart(ii):pend(ii));
+
+            % Line simplification
+            if numel(x) > 2
+                mask      = opheimSimplify(x, y, tol);
+                % Remove all those with mask==0 respecting the number of
+                % data points in the previoous segments
+                id_remove = find(mask==0) + countData;
+            end
 
-    xPixelWidth = 1/xToPix;
-    yPixelWidth = 1/yToPix;
-    tol = min(xPixelWidth,yPixelWidth);
-
-    % Split up lines which are seperated by NaNs
-    inan   = isnan(vxData) | isnan(vyData);
-    df     = diff([false, ~inan, false]);
-    pstart = find(df == 1);
-    pend   = find(df == -1)-1;
-    nlines = numel(pstart);
-
-    [linesx, linesy] = deal(cell(1,nlines*2));
-    for ii = 1:nlines
-        % Visual data used for simplifications
-        vx = vxData(pstart(ii):pend(ii));
-        vy = vyData(pstart(ii):pend(ii));
-
-        % Actual data that inherits the simplifications
-        x = xData(pstart(ii):pend(ii));
-        y = yData(pstart(ii):pend(ii));
-
-        % Line simplification
-        if numel(vx) > 2
-            mask = opheimSimplify(vx,vy,tol);
-            x    = x(mask);
-            y    = y(mask);
+            % Update the number of processed data points
+            countData = countData + pend(ii) - pstart(ii);
         end
+    end
 
-        % Place eventually simplified lines segments on odd positions
-        linesx{ii*2-1} = x;
-        linesy{ii*2-1} = y;
+    % Remove the data points
+    removeData(meta, handle, id_remove)
+end
+% =========================================================================
+function mask = isInBox(data, xLim, yLim)
 
-        % Add nans back (if any) in between the line segments
-        linesx{ii*2} = nan;
-        linesy{ii*2} = nan;
-    end
-    xData = [linesx{1:end-1}];
-    yData = [linesy{1:end-1}];
+  mask = data(:,1) > xLim(1) & data(:,1) < xLim(2) ...
+       & data(:,2) > yLim(1) & data(:,2) < yLim(2);
+end
+% =========================================================================
+function mask = segmentVisible(data, dataIsInBox, xLim, yLim)
+    % Given a bounding box {x,y}Lim, determine whether the line between all 
+    % pairs of subsequent data points crosses the box.
```

Comment:
A bit cryptic, I would suggest a more generic H1 line, the details are left inside the code